### ISSUE TITLE
Gradient descent (logistic function)

### ISSUE THREAD

**Issue by ag88** [digorithm/ArtificialIntelligenceAlgorithms#1]
your article on gradient descent is cool ;)
http://rodrigoaraujo.me/articles/2015-01/Making-Your-Machine-Think-Learn-And-Predict-Gradient-Descent-Algorithm-in-Java

i hacked together some sigmoid functions 
```
	public static double sigmoid(double x, double theta0, double theta1) {
		return 1.0D / ( 1.0D + Math.exp(-1.0D * (theta1*x + theta0)));				
	}

	public static double sigmoid(double x) {
		return 1.0D / ( 1.0D + Math.exp(-x));				
	}
```
and used them as the hypothesis function, the function with multiple parameters is useful for plotting the graph
then i used the data in this wikipedia example
https://en.wikipedia.org/wiki/Logistic_regression#Probability_of_passing_an_exam_versus_hours_of_study

surprisingly it works and converges, i'm able to use a learning rate (alpha) of 0.1!
then every 200 iterations i reduce the learning rate ``alpha *= 0.9; ``
i edited codes so that it compute the change in theta (theta - theta_old) and managed to get it to converge to 1e-4
that gives a rather fast convergence in about 1600 iterations
i got the coefficients pretty close to the wikipedia log regression example, it is probably more accurate :D 
cheers
